src/draw/lv_img_src.c

```diff
@@ -0,0 +1,191 @@
+/**
+ * @file lv_img.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "lv_img_src.h"
+#include "lv_img_buf.h"
+
+#include "../misc/lv_assert.h"
+
+static lv_res_t alloc_str_src(lv_img_src_t * src, const char * str);
+
+static const char * find_ext(const char * src);
+
+/**********************
+ * GLOBAL PROTOTYPES
+ **********************/
+
+/**
+ * Get the type of an image source
+ * @param src pointer to an image source:
+ *  - pointer to an 'lv_img_t' variable (image stored internally and compiled into the code)
+ *  - a path to a file (e.g. "S:/folder/image.bin")
+ *  - or a symbol (e.g. LV_SYMBOL_CLOSE)
+ * @return type of the image source LV_IMG_SRC_VARIABLE/FILE/SYMBOL/UNKNOWN
+ * @deprecated You shouldn't rely on this function to find out the image type
+ */
+lv_img_src_type_t lv_img_src_get_type(const void * src)
+{
+    lv_img_src_type_t img_src_type = LV_IMG_SRC_UNKNOWN;
+
+    if(src == NULL) return img_src_type;
+    const uint8_t * u8_p = src;
+
+    /*The first byte shows the type of the image source*/
+    if(u8_p[0] >= 0x20 && u8_p[0] <= 0x7F) {
+        img_src_type = LV_IMG_SRC_FILE; /*If it's an ASCII character then it's file name*/
```

Comment:
That's a copy and paste of the current code in LVGL. I *guess* it's because of the LV_LOG_WARN macro below that make sure at least one source type is detected. But, since there's a `else` in the if/else chain, the log can't be triggered. 
I'll remove it.